server/index.js

```diff
@@ -1,93 +1,79 @@
-  if (err) console.log(err);
-  else console.log("db connect success");
+const db = new sqlite3.Database('./database/productHeader.db', err => {
+	if (err) console.log(err);
+	else console.log('db connect success');
 });
 
 // query DB
-app.get("/api/items/:id", (req, res) => {
-  // sends client product table
-  db.get(
-    "SELECT * FROM product WHERE id=(?)",
-    [req.params.id],
-    (err, response) => {
-      if (err) console.log("db get request failed:", err);
-      else {
-        res.send(response);
-      }
-    }
-  );
+app.get('/api/items/:id', (req, res) => {
+	// sends client product table
+	db.get('SELECT * FROM product WHERE id=(?)', [req.params.id], (err, response) => {
+		if (err) console.log('db get request failed:', err);
+		else {
+			res.send(response);
+		}
+	});
 });
 
-app.get("/api/images/:id", (req,res) => {
-  db.all(
-    "SELECT * FROM img WHERE id_product=(?)",
-    [req.params.id],
-    (err, response) => {
-      if (err) console.log("db get img request failed:", err);
-      else {
-        res.send(response)
-      }
-    }
-  );
+app.get('/api/images/:id', (req, res) => {
+	db.all('SELECT * FROM img WHERE id_product=(?)', [req.params.id], (err, response) => {
+		if (err) console.log('db get img request failed:', err);
+		else {
+			res.send(response);
+		}
+	});
 });
 
-app.get("/api/description/:id", (req,res) => {
-  db.all("SELECT * FROM description WHERE id_product=(?)",
-  [req.params.id],
-  (err, response) => {
-    if (err) console.log("db get description request failed:", err);
-    else {
-      res.send(response)
-    }
-  })
-})
-
-app.get("/api/category/:id", (req,res) => {
-  db.all("SELECT * FROM category WHERE id_product=(?)",
-  [req.params.id],
-  (err, response) => {
-    if (err) console.log("db get category request failed:", err);
-    else {
-      res.send(response)
-    }
-  })
-})
+app.get('/api/description/:id', (req, res) => {
+	db.all('SELECT * FROM description WHERE id_product=(?)', [req.params.id], (err, response) => {
+		if (err) console.log('db get description request failed:', err);
+		else {
+			res.send(response);
+		}
+	});
+});
 
-app.get("/api/option_categories/:id", (req,res) => {
-  db.all("SELECT * FROM optionCategories WHERE id_category=(?)",
-  [req.params.id],
-  (err, response) => {
-    if (err) console.log("db get optionCategories request failed:", err);
-    else {
-      res.send(response)
-    }
-  })
-})
+app.get('/api/category/:id', (req, res) => {
+	db.all('SELECT * FROM category WHERE id_product=(?)', [req.params.id], (err, response) => {
+		if (err) console.log('db get category request failed:', err);
+		else {
+			res.send(response);
+		}
+	});
+});
 
+app.get('/api/option_categories/:id', (req, res) => {
+	db.all('SELECT * FROM optionCategories WHERE id_category=(?)', [req.params.id], (err, response) => {
+		if (err) console.log('db get optionCategories request failed:', err);
+		else {
+			res.send(response);
+		}
+	});
+});
 
```

Comment:
This file contains a lot of very similar db calls. Perhaps there is a way that you can use one call and pass it different arguments based on the call.

Reply:
Legacy code. I could change it to 1 request to get all data, but that would require modifying frontend.